go-controller/pkg/node/gateway_localnet.go

```diff
@@ -119,11 +119,7 @@ func (l *localPortWatcher) AddService(svc *kapi.Service) {
 }
 
 func (l *localPortWatcher) UpdateService(old, new *kapi.Service) {
-	if reflect.DeepEqual(new.Spec.Ports, old.Spec.Ports) &&
-		reflect.DeepEqual(new.Spec.ExternalIPs, old.Spec.ExternalIPs) &&
-		reflect.DeepEqual(new.Spec.ClusterIP, old.Spec.ClusterIP) &&
-		reflect.DeepEqual(new.Spec.Type, old.Spec.Type) &&
-		reflect.DeepEqual(new.Status.LoadBalancer.Ingress, old.Status.LoadBalancer.Ingress) {
+	if apiequality.Semantic.DeepEqual(new.Spec, old.Spec) && apiequality.Semantic.DeepEqual(new.Status, old.Status) {
```

Comment:
same

Reply:
And as Tim pointed out, this was specifically testing only the fields we actually care about, to avoid extraneous deletes+adds.

Reply:
I recover it